src/datapoints-graph/charts/charts.component.ts

```diff
@@ -200,6 +202,38 @@ export class ChartsComponent implements OnChanges, OnInit, OnDestroy {
     }
   }
 
+  async exportAsCsv() {
+    this.chartOption$
+      .pipe(
+        first(),
+        map(({ series }) => {
+          if (!Array.isArray(series)) {
+            series = [series];
+          }
+
+          const csvData = series
+            .map(({ data, name, id }) => {
+              if (Array.isArray(data)) {
+                data.unshift([
+                  'Date',
+                  `"${name} ${
+                    id.toString().split('/').length === 2
+                      ? `(${id.toString().split('/')[1]})`
+                      : ''
+                  }"`,
+                ]);
+                return data;
+              }
+            })
+            .filter(Boolean);
+          return csvData;
+        })
+      )
+      .subscribe((csvData) => {
+        this.exportCsvService.exportToCsv(csvData);
+      });
+  }
```

Comment:
Shouldn't be another datapoint another column? instead of another table?
![image](https://github.com/SoftwareAG/cumulocity-community-plugins/assets/23506385/bb90be2d-cdcd-45c3-a400-e3b76c89dfbc)

Reply:
this would mean we would need to merge the dates and fill up empty states with empty. In your case this would be okay, but in most cases, it is not that nice... So I decided to go with two and anyone can merge it on their own.